src/cpr_sdk/vespa.py

```diff
@@ -104,6 +104,17 @@ def build_vespa_request_body(parameters: SearchParameters) -> dict[str, str]:
             "input.query(query_embedding)"
         ] = "embed(msmarco-distilbert-dot-v5, @query_string)"
 
+    if parameters.custom_vespa_request_body is not None:
+        overlapping_keys = set(vespa_request_body.keys()) & set(
+            parameters.custom_vespa_request_body.keys()
+        )
+        if overlapping_keys:
+            _LOGGER.warning(
+                f"Custom request body contains overlapping keys that will override defaults: {overlapping_keys}"
+            )
```

Comment:
This overlapping keys check feels like it could be lifted into a distinct utility method, that would also make it easy to unit test

Reply:
An important part of this to me is specifying that the _request body_ contains overlapping keys, which I think we'd then loose if lifting out to a utility method? Wdyt?